src/main/java/redis/clients/jedis/DefaultJedisSocketFactory.java

```diff
@@ -64,6 +70,25 @@ public DefaultJedisSocketFactory(HostAndPort hostAndPort, JedisClientConfig conf
     }
   }
 
+  private void connectToFirstSuccsefulHost(Socket socket, HostAndPort hostAndPort) throws IOException {
+    List<InetAddress> hosts = Arrays.asList(InetAddress.getAllByName(hostAndPort.getHost()));
+    Collections.shuffle(hosts);
+    List<ConnectException> connectExceptions = new ArrayList<ConnectException>();
```

Comment:
`ConnectException` can be use `Exception` ? because connect will throw: IOException, SocketTimeoutException, java.nio.channels.IllegalBlockingModeException, IllegalArgumentException

Reply:
sure np

Reply:
> sure np

@adiamzn What is that mean? 

Should we replace `ConnectException` with `Exception` ?

Reply:
ohh sorrry :)

it means sure no problem

Reply:
@adiamzn Does `no problem` mean that it is correct to use `ConnectException`?

But if `socket.connect()` throw `SocketTimeoutException`, it will not be caught here.

Reply:
I apologize for not being clear.

I changed it to Exception instead of ConnectException.
I agree with your comment

Reply:
@adiamzn https://github.com/redis/jedis/blob/master/src/main/java/redis/clients/jedis/DefaultJedisSocketFactory.java#L102

here, I think we should also change to `Exception`, WDYT？ if you agree, pls also update it.